src/context/WalletProvider/KeepKey/hooks/useKeepKeyVersions.ts

```diff
@@ -0,0 +1,83 @@
+import { isKeepKey, KeepKeyHDWallet } from '@shapeshiftoss/hdwallet-keepkey'
+import axios from 'axios'
+import { useEffect, useState } from 'react'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+interface VersionUrl {
+  version: string
+  url: string
+}
+
+export interface FirmwareDetails {
+  bootloader: VersionUrl
+  firmware: VersionUrl
+}
+
+interface FirmwareReleases {
+  latest: FirmwareDetails
+  hashes: {
+    bootloader: Record<string, string>
+    firmware: Record<string, string>
+  }
+}
+
+interface VersionStatus {
+  device: string
+  latest: string
+  updateAvailable: boolean
+}
+
+interface Versions {
+  bootloader: VersionStatus
+  firmware: VersionStatus
+}
+
+export const useKeepKeyVersions = () => {
+  const [versions, setVersions] = useState<Versions>()
+  const { state } = useWallet()
+  const { wallet } = state
+
+  const getBootloaderVersion = (keepKey: KeepKeyHDWallet, releases: FirmwareReleases): string => {
+    const hash = keepKey.features?.bootloaderHash.toString() ?? ''
+    const buffer = Buffer.from(hash, 'base64')
+    const hex = buffer.toString('hex')
+    return releases.hashes.bootloader[hex.toLowerCase()]
+  }
+
+  useEffect(() => {
+    if (!wallet || !isKeepKey(wallet)) return
+    ;(async () => {
+      const { data: releases } = await axios.get<FirmwareReleases>(
+        // We'll need to update this whenever new firmware/bootloader is released.
+        'https://bafybeied24gc2ipvlxdbs4v676dwho2l5aafmngrleic3do2czdvgb546u.ipfs.dweb.link/keepKey.json',
```

Comment:
You should put this into `config.ts`.

Reply:
Added as config in https://github.com/shapeshift/web/pull/1341/commits/47f1a601be3bc9b889045bc85bbbbdef4f9b43e7.